### PR TITLE
Add book logic

### PR DESCRIPTION
**Pull request by sategie**
Added logic for uploading a new book, preventing duplicate books and showing an image preview before uploading an image.

Included the AddBook component in the navbar.

Added some styling to the AddBook component.

### CODE REVIEW

**Review by nora-kauczor — APPROVED** [review]

File: bookrack-backend/src/test/java/org/example/bookrackbackend/BookControllerTest.java [hunk 1]
```diff
@@ -95,14 +95,6 @@ void getBookById_shouldThrowException_whenIdNotFound() throws Exception {
...
     }
 
-    @Test
-    void getBookById_shouldThrowException_whenInvalidIdProvided() throws Exception {
-        mockMvc.perform(MockMvcRequestBuilders.get("/api/books/#")).andExpect(status().isNotFound())
-                .andExpect(content().json("""
-                {"message": "Please check entered URL."}
-                """));
-
-    }
```
Comment (sieberss) [inline bookrack-backend/src/test/java/org/example/bookrackbackend/BookControllerTest.java:105]: Why have you removed this test?

A check for invalid ID seems sensible, but I would expect another response message.

File: bookrack-backend/src/test/java/org/example/bookrackbackend/BookControllerTest.java [hunk 2]
```diff
@@ -124,4 +116,35 @@ void addBook_shouldReturnSuccess_whenAllFieldsAreProvided() throws Exception {
                         .param("year", "2021"))
                 .andExpect(status().isOk());
     }
+
+    @DirtiesContext
+    @Test
+    void addBook_shouldReturnBadRequest_whenBookWithSameTitleAndAuthorExists() throws Exception {
+        when(cloudinary.uploader()).thenReturn(uploader);
+        when(uploader.upload(any(), any())).thenReturn(Map.of("url", "test"));
```
Comment (sieberss) [inline bookrack-backend/src/test/java/org/example/bookrackbackend/BookControllerTest.java:124]: what do you use the returned Map for? I can't find "url" and "test" in your test data. 

Maybe these lines are superflous or you should check the Map entries. (when-commands usually appear in unit tests, rarely in integration tests.)

File: bookrack-backend/src/test/java/org/example/bookrackbackend/BookServiceTest.java [hunk 3]
```diff
@@ -90,18 +90,74 @@ void addBook_shouldAddBookToDatabase() throws IOException {
...
+        when(mockCloudinaryService.uploadImage(image)).thenReturn(expectedImageUrl);
 
+        Book expectedSavedBook = new Book(
+                null,
+                bookCreationDTO.title(),
+                bookCreationDTO.author(),
+                bookCreationDTO.country(),
+                parseInt(bookCreationDTO.year()),
+                expectedImageUrl
+        );
```
Comment (sieberss) [inline bookrack-backend/src/test/java/org/example/bookrackbackend/BookServiceTest.java:116]: where do you check that addBook generates a new ID?

**Review by sieberss — APPROVED** [review]
Looks good so far.

But please check if your tests are ok.

**Pull request merged**